[init] Board 기본 파일 생성 (Board 참여자 작업하지 않음)

ayboori opened the pull request:
## 관련 Issue

<!-- 해당 Pull Request와 관련된 Issue를 적습니다. -->

* #14 

## 변경 사항

<!-- 이 Pull Request에서 어떤 점이 변경되었는지 간단하게 설명해주세요.
화면을 첨부한 설명이 필요한 경우 스크린샷을 첨부해 주세요 -->

* 보드 전체 조회
* 보드 1개 조회
* 보드 수정 (사용자 검증 후 유효한 사용자일 경우에만 수행)
* 보드 삭제 (사용자 검증 후 유효한 사용자일 경우에만 수행)

## Todo List

<!-- 이번 Pull Request 작업에서 아직 처리하지 못한 작업이나  
    추후에 해결해야 될 문제들을 기입해 주세요 -->  

* 테스트 코드 작성 전입니다. (PR 올린 후 작성해보겠습니다!)
* 아직 포스트맨 테스트를 해보지 않았습니다.
* boardMemebers 관련 기능 구현되어 있지 않습니다. (희열님 작업하시는 것으로 알고 있습니다)
* Deck과의 연관 관계 선언 되어있지 않습니다 (deck 구현 이후 수정 예정)
* Color 부분 프론트에서 작업할 예정인지, 백엔드 단에서 따로 해야 할 작업이 있는지 궁금합니다. (현재 String 타입으로 필드 선언만 되어 있음)

## Check List

<!-- 기능 구현을 다 했다면? --> 

- [ ] 포스트맨으로 체크해 보았나요?
- [ ] 테스트 코드를 작성하셨나요?

ayboori committed:
[init] Board 초기 설정

ayboori committed:
Merge branch 'dev' into feature/board

ayboori committed:
줄 바꿈 정리

proLmpa commented:
Color 부분은 프론트에서 작업할 거라 백에서는 그냥 String 타입으로 두셔도 됩니다!

ayboori committed:
오류 수정 중

Wooin-dev commented on src/main/java/com/example/itwassummer/board/controller/BoardController.java:
```diff
@@ -0,0 +1,63 @@
+@RequiredArgsConstructor
+@Tag(name = "보드 API", description = "보드 기능과 관련된 API 정보를 담고 있습니다.")
+public class BoardController {
+    private final BoardServiceImpl boardService;
```
DB오류 때문에 고생많으셔요!😂

BoardServiceImpl 대신에  BoardService 인터페이스를 멤버로 선언해주고
BoardServiceImpl에 @Service를 달아주면 Impl을 주입받아올 수 있다고 합니다!

여기 이 부분을 비슷하게 희열님께서 질문해주신적이 있어서 슬랙 링크도 남겨요
[https://nbcamp.slack.com/archives/C058CRUFXAM/p1690633441178169](url)

Wooin-dev commented on src/main/java/com/example/itwassummer/board/service/BoardServiceImpl.java:
```diff
@@ -0,0 +1,97 @@
+
+    // 단일 보드 조회
+    @Override
+    public BoardResponseDto showBoards(Long id) {
```
조회시에는 
@Transactional (readOnly = true) 를 달아주자고 하더라고요!
생성시에도 메소드 전체가 완전히 실행되어야 하는 경우엔(중간에 예외가 발생하는 경우를 대비) 롤백기능을 위해 @Transactional을 달아두는 선택지도 있을 것 같습니다 :)

Wooin-dev reviewed (approved):
DB땜에 힘드신 와중 고생많으셔요.!

proLmpa committed:
Delete settings.gradle

rootProject.name은 해당 프로젝트가 존재하는 폴더 명을 적어야 하는데 이게 지금 팀원마다 다를 겁니다.

그래서 해당 파일은 현재 PR에서 제거하겠습니다.
그리고 .gitignore에 이 파일도 포함 시키는 게 좋겠네요

proLmpa committed:
Merge branch 'dev' into feature/board
